### PR TITLE
Merge hotfix 10.0.6 branch to stable10

### PR DESCRIPTION
**Pull request by PVince81**
Hotfix for 10.0.6 was built, merging back changes to stable10

### CODE REVIEW

**Comment by PVince81** [comment]
Rebasing a hotfix branch feels wrong.

Will likely need to push a new commit to solve the conflict before merging.

**Comment by PVince81** [comment]
Merged stable10 into this branch. Solved conflicts, regenerated composer.lock.

@DeepDiver1975 please review

**Comment by PVince81** [comment]
Ouch, what's going on ??
```
12:18:36 Your requirements could not be resolved to an installable set of packages.
12:18:36 
12:18:36   Problem 1
12:18:36     - Installation request for doctrine/annotations v1.6.0 -> satisfiable by doctrine/annotations[v1.6.0].
12:18:36     - doctrine/annotations v1.6.0 requires php ^7.1 -> your PHP version (7.0.15) does not satisfy that requirement.
12:18:36   Problem 2
12:18:36     - Installation request for doctrine/cache v1.7.1 -> satisfiable by doctrine/cache[v1.7.1].
12:18:36     - doctrine/cache v1.7.1 requires php ~7.1 -> your PHP version (7.0.15) does not satisfy that requirement.
12:18:36   Problem 3
12:18:36     - Installation request for doctrine/collections v1.5.0 -> satisfiable by doctrine/collections[v1.5.0].
12:18:36     - doctrine/collections v1.5.0 requires php ^7.1 -> your PHP version (7.0.15) does not satisfy that requirement.
12:18:36   Problem 4
12:18:36     - Installation request for doctrine/common v2.8.1 -> satisfiable by doctrine/common[v2.8.1].
12:18:36     - doctrine/common v2.8.1 requires php ~7.1 -> your PHP version (7.0.15) does not satisfy that requirement.
12:18:36   Problem 5
12:18:36     - Installation request for doctrine/dbal v2.6.3 -> satisfiable by doctrine/dbal[v2.6.3].
12:18:36     - doctrine/dbal v2.6.3 requires php ^7.1 -> your PHP version (7.0.15) does not satisfy that requirement.
12:18:36   Problem 6
12:18:36     - Installation request for doctrine/inflector v1.3.0 -> satisfiable by doctrine/inflector[v1.3.0].
12:18:36     - doctrine/inflector v1.3.0 requires php ^7.1 -> your PHP version (7.0.15) does not satisfy that requirement.
12:18:36   Problem 7
12:18:36     - Installation request for symfony/debug v4.0.4 -> satisfiable by symfony/debug[v4.0.4].
12:18:36     - symfony/debug v4.0.4 requires php ^7.1.3 -> your PHP version (7.0.15) does not satisfy that requirement.
12:18:36   Problem 8
12:18:36     - Installation request for doctrine/instantiator 1.1.0 -> satisfiable by doctrine/instantiator[1.1.0].
12:18:36     - doctrine/instantiator 1.1.0 requires php ^7.1 -> your PHP version (7.0.15) does not satisfy that requirement.
12:18:36   Problem 9
12:18:36     - Installation request for ocramius/proxy-manager 2.1.1 -> satisfiable by ocramius/proxy-manager[2.1.1].
12:18:36     - ocramius/proxy-manager 2.1.1 requires php ^7.1.0 -> your PHP version (7.0.15) does not satisfy that requirement.
12:18:36   Problem 10
12:18:36     - Installation request for symfony/config v4.0.4 -> satisfiable by symfony/config[v4.0.4].
12:18:36     - symfony/config v4.0.4 requires php ^7.1.3 -> your PHP version (7.0.15) does not satisfy that requirement.
12:18:36   Problem 11
12:18:36     - Installation request for symfony/dependency-injection v4.0.4 -> satisfiable by symfony/dependency-injection[v4.0.4].
12:18:36     - symfony/dependency-injection v4.0.4 requires php ^7.1.3 -> your PHP version (7.0.15) does not satisfy that requirement.
12:18:36   Problem 12
12:18:36     - Installation request for symfony/filesystem v4.0.4 -> satisfiable by symfony/filesystem[v4.0.4].
12:18:36     - symfony/filesystem v4.0.4 requires php ^7.1.3 -> your PHP version (7.0.15) does not satisfy that requirement.
12:18:36   Problem 13
12:18:36     - Installation request for symfony/yaml v4.0.4 -> satisfiable by symfony/yaml[v4.0.4].
12:18:36     - symfony/yaml v4.0.4 requires php ^7.1.3 -> your PHP version (7.0.15) does not satisfy that requirement.
12:18:36   Problem 14
12:18:36     - Installation request for zendframework/zend-code 3.3.0 -> satisfiable by zendframework/zend-code[3.3.0].
12:18:36     - zendframework/zend-code 3.3.0 requires php ^7.1 -> your PHP version (7.0.15) does not satisfy that requirement.
12:18:36   Problem 15
12:18:36     - symfony/debug v4.0.4 requires php ^7.1.3 -> your PHP version (7.0.15) does not satisfy that requirement.
12:18:36     - symfony/console v3.4.4 requires symfony/debug ~2.8|~3.0|~4.0 -> satisfiable by symfony/debug[v4.0.4].
12:18:36     - Installation request for symfony/console v3.4.4 -> satisfiable by symfony/console[v3.4.4].
```

**Comment by PVince81** [comment]
Composer works for me locally on this branch and also on stable10

@DeepDiver1975 please help, dependencies master!

**Comment by PVince81** [comment]
ok, reproducible with PHP 5.6

**Comment by PVince81** [comment]
My mistake was running `make update-composer` on PHP 7.1... I ran it again on PHP 5.6 and now it looks good.

Please double check.

File: composer.lock [hunk 1]
```diff
@@ -2323,16 +2328,16 @@
         },
         {
             "name": "sabre/vobject",
-            "version": "4.1.2",
+            "version": "4.1.4",
```
Comment (phil-davis) [inline composer.lock]: It is likely that this sabre/vobject bump will also need unit test changes that are in:
https://github.com/owncloud/core/pull/30266/commits/c3dc33c6eaafefa78637bc395855ba4086925f20

**Comment by PVince81** [comment]
damn... ok I'll see if I can make it only update the one library instead...

**Comment by phil-davis** [comment]
Actually IMO this is "a good thing" to get ``stable10`` up to date with all the minor version bumps of stuff that is waiting to happen.

**Comment by phil-davis** [comment]
Of course it is not "the same" as what happened in the hotfix branch, but actually that does not matter so much. If you can just cherry-pick in the sabre/vobject unit test changes and see if CI passes...

**Comment by PVince81** [comment]
> Actually IMO this is "a good thing" to get stable10 up to date with all the minor version bumps of stuff that is waiting to happen.

Yes, but I'd prefer for this to happen on another branch, not in the context of the minor changes done in this branch.

**Comment by PVince81** [comment]
```
vincent@nohostname:~/work/workspace/owncloud> composer update symfony/translation
Loading composer repositories with package information
Updating dependencies (including require-dev)
Package operations: 0 installs, 46 updates, 0 removals
  - Updating doctrine/cache (v1.6.2 => v1.6.1): Loading from cache
  - Updating doctrine/common (v2.7.3 => v2.7.2): Loading from cache
  - Updating doctrine/dbal (v2.5.13 => v2.5.12): Loading from cache
  - Updating symfony/routing (v3.4.4 => v3.2.4): Downloading (100%)         
  - Updating symfony/process (v3.4.4 => v3.2.4): Downloading (100%)         
  - Updating swiftmailer/swiftmailer (v5.4.9 => v5.4.8): Loading from cache
...
```

http://memegenerator.net/instance/81211641

**Comment by PVince81** [comment]
ok let's hope it only updated a minimum set now

**Comment by PVince81** [comment]
oh look... the diff of composer.lock is empty, so this now matches stable10...

**Comment by phil-davis** [comment]
The ``symfony/translation`` stuff was merged to ``stable10`` a week ago:
https://github.com/owncloud/core/commit/62c6e29b13f1e7c5f0af4729c712bc4dac411a63#diff-b5d0ee8c97c7abd7e3fa29b9a27d1780
(that is why there are now no diffs at all - not even of ``composer.json``)

So actually all the commits here related to ``composer.*`` could be dropped out - or make a new PR that just cherry-picks the version changes and changelog entries. That would make a cleaner history.

**Comment by PVince81** [comment]
> So actually all the commits here related to composer.* could be dropped out 

the branch itself had conflicts with stable10, that's why I had to use a merge commit and resolve them.

Let's see how far we go with this. Worst case like you say I made a new branch and cherry-pick only the needed things.

**Comment by phil-davis** [comment]
It took 2 minutes to make #30329 which has just the relevant commits. See if you like that.

**Comment by PVince81** [comment]
Superseded by https://github.com/owncloud/core/pull/30329.

Please don't delete the branch!

**Pull request closed without merging**